src/charm.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+"""Subordinate charm for TimescaleDB."""
+import os
+from subprocess import PIPE, Popen, check_call, check_output
+
+from ops.charm import CharmBase
+from ops.framework import StoredState
+from ops.main import main
+from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
+
+
+class TimescaleDB(CharmBase):
+    """Subordinate charm for TimescaleDB."""
+
+    state = StoredState
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.framework.observe(self.on.install, self._on_install)
+        self.framework.observe(self.on.upgrade_charm, self._on_upgrade_charm)
+
+    def _on_install(self, event):
+        if not hasattr(self.state, "installed"):
```

Comment:
I'd go with
```
installed = getattr(self.state, "installed", False)
if installed:
    return
```
But could we check whether the package is installed by calling dpkg instead, and avoid state completely?